### DISCUSSION TITLE
[Form components] Change `size` prop default value to `fillParent`

### DISCUSSION THREAD

**Discussion by GomezIvann** [dxc-technology/halstack-react#1031] [Ideas]
After seeing other design systems I think `fillParent` suits better our form components, since we now can know in which containers they will be fitted (Layout components) and predict that `"medium"` is the most used size, in my opinion, is a bit pretentious. This change would affect:

1. Text input, Number input, Password input, Date input and Textarea.
2. Select.
3. ...?

Approaches:

- Kaizen: https://cultureamp.design/components/text-field/
- Braid: https://seek-oss.github.io/braid-design-system/components/TextField
- Shopify: https://polaris.shopify.com/components/forms/form#navigation
- Ant: https://ant.design/components/input/
- Atlassian: https://atlassian.design/components/textfield/examples

**Comment by marcialfps** [comment]
I fully agree. If we provide layout components we should at least delete fixed sizes. 

Also, I think Alex had the idea of deleting the `size` prop from all components, so the best idea is to continue with that goal.
